components/scream/src/share/grid/remap/abstract_remapper.hpp

```diff
@@ -218,10 +219,15 @@ class AbstractRemapper
   int           m_num_fields;
   int           m_num_registered_fields;
 
-  // This vector stores a boolean whose elements are the indices of fields that
-  // have been bound to the remapper. This vector is necessary because the
-  // binding of fields is separate from their regіstration: recall that one may
-  // register a field using its identifier, and bind the actual field later.
+  // This vector maps the indices of registered fields to booleans that indicate
+  // whether these fields have been bound to the remapper. This vector is
+  // necessary because the binding of fields is separate from their
+  // regіstration: recall that one may register a field using its identifier,
+  // and bind the actual field later.
+  // NOTE: vector<bool> is a strange beast, and doesn't necessary behave as
+  // NOTE: expected. Use caution when manipulating this member, and don't
```

Comment:
behave 'as' expected.

Reply:
And is 'NOTE:' on all lines intended?

Reply:
I did intend that with NOTE:, but have changed. `as` was already there, but I changed `necessary` to `necessarily`.